src/webgpu/util/f32_interval.ts

```diff
@@ -702,15 +829,28 @@ const DivisionIntervalOp: BinaryToIntervalOp = {
   extrema: (x: F32Interval, y: F32Interval): [F32Interval, F32Interval] => {
     // division has a discontinuity at y = 0.
     if (y.contains(0)) {
-      y = toInterval(0);
+      y = toF32Interval(0);
     }
     return [x, y];
   },
 };
 
 /** Calculate an acceptance interval of x / y */
 export function divisionInterval(x: number | F32Interval, y: number | F32Interval): F32Interval {
-  return runBinaryOp(toInterval(x), toInterval(y), DivisionIntervalOp);
+  return runBinaryOp(toF32Interval(x), toF32Interval(y), DivisionIntervalOp);
+}
+
+const DotIntervalOp: VectorPairToIntervalOp = {
+  impl: (x: number[], y: number[]): F32Interval => {
+    // dot(x, y) = sum of x[i] * y[i]
+    const multiplications: F32Interval[] = x.map((_, i) => multiplicationInterval(x[i], y[i]));
```

Comment:
500+ lines of changes, so I can write this out succinctly in a functional manner